src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -386,6 +513,55 @@ public override void InvokeCallback(bool allowPooling)
 
                 cb(bt, sa, sal, SocketFlags.None, ec);
             }
+
+            internal override unsafe bool TryBatch(SocketAsyncContext context, int id, ref Interop.Sys.IoControlBlock ioControlBlock)
+            {
+                if (IsSync || !TrySetRunning())
+                {
+                    return false;
+                }
+
+                PinHandle = Buffer.Pin();
+
+                ioControlBlock.AioData = (ulong)id;
```

Comment:
minor: the order of field initializaiton is the same as they are lay out in memory to make sure we benefit from data locality